mlflow/tracing/types/wrapper.py

```diff
@@ -106,14 +100,37 @@ def set_attributes(self, attributes: Dict[str, Any]):
                 f"Attributes must be a dictionary, but got {type(attributes)}. Skipping."
             )
             return
-        self._attributes.update(attributes)
+
+        for key, value in attributes.items():
+            self.set_attribute(key, value)
 
     def set_attribute(self, key: str, value: Any):
         """Set a single attribute to the span."""
+        self._set_attribute(key, value)
+
+    def _set_attribute(self, key: str, value: Any, serialize=True):
```

Comment:
note: The `serialize` argument is added for avoiding unnecessary serialization for reserved string attribute like `mlflow.traceRequestId`, `mlflow.spanType`, which we already know they are string. For other attributes that accepts user's input, we always do JSON serialization. The argument is only added to the private method so it won't be used by users.

Reply:
Do we avoid serialization of those strings so UI can capture them easier? While from another perspective, if we always serialize it then UI can always deserialize the value no matter what key it is, which seems even simpler

Reply:
Yea the simplification makes sense to me. I avoided the serialization for efficiency as the `request_id` is the most frequently read property. However, actually the better way to handle this seems to be having instance attribute like `self._request_id` because it won't be updated twice.

Reply:
@daniellok-db btw is there any preference from frontend perspective?